FWCore/Framework/interface/GetterOfProducts.h

```diff
@@ -187,6 +191,54 @@ namespace edm {
       }
     }
 
+    void fillHandles(edm::EventForOutput const& event, std::vector<edm::Handle<T>>& handles) const {
+      handles.clear();
+      if (branchType_ == edm::InEvent) {
+        handles.reserve(tokens_->size());
+        for (auto const& token : *tokens_) {
+          if (auto handle = event.getHandle(token)) {
+            handles.push_back(handle);
+          }
+        }
+      }
+    }
+
+    void fillHandles(edm::LuminosityBlockForOutput const& lumi, std::vector<edm::Handle<T>>& handles) const {
+      handles.clear();
+      if (branchType_ == edm::InLumi) {
+        handles.reserve(tokens_->size());
+        for (auto const& token : *tokens_) {
+          if (auto handle = lumi.getHandle(token)) {
+            handles.push_back(handle);
+          }
+        }
+      }
+    }
+
+    void fillHandles(edm::RunForOutput const& run, std::vector<edm::Handle<T>>& handles) const {
+      handles.clear();
+      if (branchType_ == edm::InRun) {
+        handles.reserve(tokens_->size());
+        for (auto const& token : *tokens_) {
+          if (auto handle = run.getHandle(token)) {
+            handles.push_back(handle);
+          }
+        }
+      }
+    }
+
+    void fillHandles(edm::ProcessBlockForOutput const& processBlock, std::vector<edm::Handle<T>>& handles) const {
+      handles.clear();
+      if (branchType_ == edm::InProcess) {
+        handles.reserve(tokens_->size());
+        for (auto const& token : *tokens_) {
+          if (auto handle = processBlock.getHandle(token)) {
+            handles.push_back(handle);
+          }
+        }
+      }
+    }
```

Comment:
Just to note, there seems to be room for abstraction to avoid copy-paste (not necessarily to be done in this PR, but I also don't mind if you do).

Reply:
I made it a template function. Seems better. Thanks.